core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -833,8 +833,10 @@ private[spark] class BlockManager(
     logDebug("Put block %s locally took %s".format(blockId, Utils.getUsedTimeMs(startTimeMs)))
 
     // Either we're storing bytes and we asynchronously started replication, or we're storing
-    // values and need to serialize and replicate them now:
-    if (putLevel.replication > 1) {
+    // values and need to serialize and replicate them now.
+    // Should not replicate the block if its StorageLevel is StorageLevel.NONE or
+    // putting it to local is failed.
+    if (!putBlockInfo.isFailed && putLevel.replication > 1) {
```

Comment:
Right , if Block failed to stored locally , the Receiver will not get the BlockId as it looks for the BlockId in PutResult even though block got replicated . But @tdas and @harishreedharan can confirm if this indeed the issue. My point is even if Receiver can check if Blocks are replicated in this case, the Replication factor is already down to 1 and thus have higher chance of getting BlockNotFoundException and streaming jobs failed .